src/wasm/literal.cpp

```diff
@@ -2388,6 +2388,16 @@ Literal extend(const Literal& vec) {
   return Literal(result);
 }
 
+template<LaneOrder Side> Literal extendF32(const Literal& vec) {
```

Comment:
An alternative to adding a new template here would be to make the `geti32` a template parameter of `extend`, but this seems fine too.

Reply:
I initially thought of unifying the "extend" too, but SIMD is already standardized and this special case is only needed once